doc/api/deprecations.md

```diff
@@ -2439,6 +2439,20 @@ The `_channel` property of child process objects returned by `spawn()` and
 similar functions is not intended for public use. Use `ChildProcess.channel`
 instead.
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: Legacy HTTP parser
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/xxxxx
+    description: Documentation-only.
+-->
+
+Type: Documentation-only
+
+The legacy HTTP parser, used by default in versions of Node.js prior to 12.0.0,
+is deprecated.
```

Comment:
Maybe include detail about what this means for the end user (e.g. deprecating usage of `--http-parser=legacy` option) as I suspect most people probably won't even know if/how this will affect them.